apps/site/pages/en/learn/modules/package-configuration.md

```diff
@@ -0,0 +1,563 @@
+
+💡 Using `"type": "module"`[^2] paired with the `.cjs` file extension (for commonjs files) yields best results. For more information on why, see [Down the rabbit-hole](#down-the-rabbithole) and [Gotchas](#gotchas) below.
+
+**Working example**: [esm-with-cjs-distro](https://github.com/JakobJingleheimer/nodejs-module-config-examples/tree/main/packages/esm/cjs-distro)
+
+### ESM source and both CJS & ESM distribution
+
+These are "mixologist" territory.
+
+When source code is written in non-JavaScript (ex TypeScript), options can be limited due to needing to use file extension(s) specific to that language (ex `.ts`) and there may be no `.mjs` equivalent.
+
+Similar to [CJS source and both CJS & ESM distribution](#cjs-source-and-both-cjs-amp-esm-distribution), you have the same options.
+
+#### Publish only a CJS distribution with property exports
+
+The "Mojito" of packages: Tricky to make and needs good ingredients.
+
+This option is almost identical to the [CJS source with CJS & ESM distribution's property exports](#attach-named-exports-directly-onto-raw-exports-endraw-) above. The only difference is in package.json: `"type": "module"`.
+
+Only some build tools support generating this output. [Rollup](https://www.rollupjs.org/) produces compatible output out of the box when targetting commonjs. Webpack as of [v5.66.0+](https://github.com/webpack/webpack/releases/tag/v5.66.0) does with the new [`commonjs-static`](https://webpack.js.org/configuration/output/#type-commonjs-static) output type, (prior to this no commonjs options produces compatible output). It is not currently possible with [esbuild](https://esbuild.github.io/) (which produces a non-static `exports`).
+
+The working example below was created prior to Webpack's recent release, so it uses Rollup (I'll get around to adding a Webpack option too).
+
+**Working example**: [esm-with-cjs-distro](https://github.com/JakobJingleheimer/nodejs-module-config-examples/tree/main/packages/esm/dual/property-distro)
+
+```json displayName="Minimal package.json"
+{
+  "name": "esm-with-cjs-distribution",
+  "type": "module",
+  "main": "./dist/index.cjs"
+}
+```
+
+```json displayName="Advanced (verbose) package.json"
+{
+  "name": "esm-with-cjs-distribution",
+  "type": "module",
+  "engines": { "node": ">=12.22.7" },
+  "exports": {
+    ".": "./dist/index.cjs",
+    "./package.json": "./package.json"
+  }
+}
+```
+
+💡 Using `"type": "module"`[^2] paired with the `.cjs` file extension (for commonjs files) yields best results. For more information on why, see [Down the rabbit-hole](#down-the-rabbithole) and [Gotchas](#gotchas) below.
+
+#### Publish a CJS distribution with an ESM wrapper
+
+The "Pornstar Martini" of packages: There's a lot going on here. Unlike the cocktail, this is usually not the best.
```

Comment:
Oof, it is an unfortunate name (it is indeed the name of a cocktail). I can find a different drink if these drink names are kept.

Reply:
the “Long Island ice tea”?

Reply:
Long Island Ice Tea is already used 😅 I chose "Pornstar Martini" because it's surprisingly packed (whereas a Long Island is famously A,MF).

Reply:
For the sake of letting everyone here be comfortable, and as a mediator, let's keep references to anything that might be only for a targeted audience (be it alcohol or anything else) out of here.

I understand the intent, but for better or for worse, let's keep these out please 🙇

Reply:
I agree that no cocktail jokes belong here.

(Also FYI "Long Island Ice Tea" is associated with some myths about rapist traps for women in my background - not everyone knows that it would taste alcoholic enough that you probably won't be tricked, but I also know people who would think this is promoting rapist tricks that trivialize alcohol content in cocktails. In general, using references that are biased towards an audience in a way that has nothing to do with the primary topic of a technical article is just a mistake IMO).

Reply:
@ovflowd as can be seen from my previous https://github.com/nodejs/nodejs.org/pull/7229#discussion_r1864679437 a mediator is not necessary.